### PR TITLE
Resources cleanup and doc updates

### PR DESCRIPTION
**Pull request by jcagme**

### CODE REVIEW

File: Documentation/Onboarding.md [hunk 1]
```diff
@@ -20,7 +22,7 @@
        [Version.Details.xml](https://github.com/dotnet/arcade-minimalci-sample/blob/master/eng/Version.Details.xml)
        files to your eng\ folder. Adjust the version prefix and prerelease label
        as necessary.
-    5. Add dotnet-core feed to
+    5. Add dotnet-core feed and any other feeds that the repository restores NuGet packages from to
        [NuGet.config](https://github.com/dotnet/arcade-minimalci-sample/blob/master/NuGet.config).
     6. Must have a root project/solution file for the repo to build.
     7. Additional package feeds can be added to the `eng\Version.props` file, e.g.
```
Comment (tmat) [inline Documentation/Onboarding.md]: Remove this step.

File: eng/Versions.props [hunk 2]
```diff
@@ -73,12 +73,4 @@
     <MicrosoftDotNetMaestroClientVersion>1.1.0-beta.19374.3</MicrosoftDotNetMaestroClientVersion>
     <MicrosoftDotNetSwaggerGeneratorMSBuildVersion>1.0.0-beta.19372.10</MicrosoftDotNetSwaggerGeneratorMSBuildVersion>
   </PropertyGroup>
-  <PropertyGroup>
```
Comment (tmat) [inline eng/Versions.props]: Isn't this already removed?

File: Documentation/Onboarding.md (hunk 1 above)
Comment (riarenas) [inline Documentation/Onboarding.md]: Should step 7 go away?

Comment (tmat) [reply Documentation/Onboarding.md]: https://github.com/dotnet/arcade/pull/3082/commits/877eb94d17fdde14d3eb73b550e0b024406219c1#diff-9846b0eefccf8f4cf8f5cbd40d6ecca2L25-L39

File: eng/Versions.props (hunk 2 above)
Comment (jcagme) [reply eng/Versions.props]: Nope, that's the second step as specified in https://github.com/dotnet/arcade/blob/master/Documentation/RestoreSourcesUpdateStatus.md#part-2

**Comment by jcagme** [comment]
@riarenas @tmat good to merge?

**Pull request merged**